daml/splice-dso-governance-test/daml/Splice/Scripts/TestValidatorLicense.daml

```diff
@@ -195,3 +196,112 @@ test_ValidatorLivenessWeightInRunNextIssuanceD = do
       abort ("Bob should receive ~15000 times rewards. Expected within: [" <> show bobExpectedMin <> ", " <> show bobExpectedMax <> "], Got: " <> show bobReward)
 
     return ()
+
+-- Test multiple validator license weight changes can be done via single vote
+testValidatorLicenseWeightChangeBatch : Script ()
+testValidatorLicenseWeightChangeBatch = do
+  (app, _, (sv1, sv2, sv3, sv4)) <- initDevNet
+
+  validator1 <- setupValidator app "validator1"
+  validator2 <- setupValidator app "validator2"
+  validator3 <- setupValidator app "validator3"
+
+  -- Verify all validators have default weight (None)
+  [(_, license1)] <- query @ValidatorLicense validator1
+  [(_, license2)] <- query @ValidatorLicense validator2
+  [(_, license3)] <- query @ValidatorLicense validator3
+  license1.weight === None
+  license2.weight === None
+  license3.weight === None
+
+  -- Change weights for all three validators in a single vote
+  initiateAndAcceptVote app [sv1, sv2, sv3, sv4] (ARC_DsoRules (SRARC_ModifyValidatorLicenses
+    (ValidatorLicensesModification with
+      changes = [VLC_ChangeWeight with validator = validator1; weight = 5.0
+               , VLC_ChangeWeight with validator = validator2; weight = 10.0
+               , VLC_ChangeWeight with validator = validator3; weight = 0.0])))
+
+  executeAllVoteExecutionInstructions app
+
+  -- Verify all weights are updated correctly
+  [(_, license1)] <- query @ValidatorLicense validator1
+  [(_, license2)] <- query @ValidatorLicense validator2
+  [(_, license3)] <- query @ValidatorLicense validator3
+  license1.weight === Some 5.0
+  license2.weight === Some 10.0
+  license3.weight === Some 0.0
+
+  pure ()
+
+-- Test multiple validator license withdrawals can be done via single vote
+testValidatorLicenseWithdrawBatch : Script ()
+testValidatorLicenseWithdrawBatch = do
+  (app, _, (sv1, sv2, sv3, sv4)) <- initDevNet
+
+  validator1 <- setupValidator app "validator1"
+  validator2 <- setupValidator app "validator2"
+  validator3 <- setupValidator app "validator3"
+
+  -- Verify all validator licenses exist
+  [_] <- query @ValidatorLicense validator1
+  [_] <- query @ValidatorLicense validator2
+  [_] <- query @ValidatorLicense validator3
+
+  -- Withdraw all three validator licenses in a single vote
+  initiateAndAcceptVote app [sv1, sv2, sv3, sv4] (ARC_DsoRules (SRARC_ModifyValidatorLicenses
+    (ValidatorLicensesModification with
+      changes = [VLC_Withdraw with validator = validator1
+               , VLC_Withdraw with validator = validator2
+               , VLC_Withdraw with validator = validator3])))
```

Comment:
the test structure here doesn't test combined changes: withdraw and change weight. I'd suggest to merge the two tests and show that multiple of both are possible in a single batch.

Reply:
Fixed, now testing both types in a single batch